Automattic/wp-calypso
# BackupBrowser: Add checkboxes to items

## Proposed Changes

* This adds a tree structure to the state which tracks checkbox selections.
* On development, or in a live branch using `?flags=jetpack/backup-granular` you can click `Select` in the header to enable checkboxes.
* Checkboxes should behave in the following ways:
  * Selecting a parent should select all children
  * Selecting a child should update all ancestors where applicable.
    * A parent should be checked if all children are checked, unchecked if all children are checked or mixed (indicated by a grey checkbox background) if some children are selected and some are not.
  * Clicking a mixed checkbox will unselect it and update children and parents accordingly
  * Expanding and collapsing the tree structure should retain check box state

Known Issues to be fixed in future PRs
* The checkbox in the header isn't connected to the list yet.
* The number of files in the header isn't connected to the list yet.
* When in select mode there will be two primary CTA buttons.

## Testing Instructions



* On development, or in a live branch using `?flags=jetpack/backup-granular` you can click `Select` in the header to enable checkboxes.
* Go through the above described ways checkboxes should behave and ensure they do.
* Checkboxes should not be able to be made visible without adding the `flags` to the URL or by being in development mode (running locally).
* 

## Pre-merge Checklist
